card/lib/card/query/run.rb

```diff
@@ -72,7 +76,7 @@ def alter_results?
 
       def card_result record, _field
         if alter_results?
-          Card.fetch process_name(record["name"])
+          Card.fetch alter_result(record["name"])
         else
           fetch_or_instantiate record
```

Comment:
Is there a reason that we don't cache when we alter the result?

Reply:
#fetch does cache.  but we can't instantiate an altered result.  (all the information needed for normal instantiation is in the record, but the record isn't sufficient for an altered result